src/components/NewCommentForm.tsx

```diff
@@ -1,69 +1,129 @@
-import React from 'react';
+import { useState } from 'react';
+import classNames from 'classnames';
+import { Input } from './Input';
+import { CommentData } from '../types/Comment';
 
-export const NewCommentForm: React.FC = () => {
-  return (
-    <form data-cy="NewCommentForm">
-      <div className="field" data-cy="NameField">
-        <label className="label" htmlFor="comment-author-name">
-          Author Name
-        </label>
+type FormData = {
+  name: string;
+  email: string;
+  body: string;
+};
 
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="name"
-            id="comment-author-name"
-            placeholder="Name Surname"
-            className="input is-danger"
-          />
+type FormErrors = {
+  name: boolean;
+  email: boolean;
+  body: boolean;
+};
 
-          <span className="icon is-small is-left">
-            <i className="fas fa-user" />
-          </span>
+type NewCommentFormProps = {
+  postId?: number;
+  handleAddComment: (data: any, url: string, msg: string) => Promise<void>;
+};
 
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
-        </div>
+const initialFormData: FormData = {
+  name: '',
+  email: '',
+  body: '',
+};
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Name is required
-        </p>
-      </div>
+const initialFormErrors: FormErrors = {
+  name: false,
+  email: false,
+  body: false,
+};
 
-      <div className="field" data-cy="EmailField">
-        <label className="label" htmlFor="comment-author-email">
-          Author Email
-        </label>
+export const NewCommentForm = ({
+  postId,
+  handleAddComment,
+}: NewCommentFormProps) => {
+  const [formData, setformData] = useState<FormData>(initialFormData);
+  const [formErrors, setformErrors] = useState<FormErrors>(initialFormErrors);
+  const [loading, setLoading] = useState(false);
```

Comment:
Consider changing `setformData` to `setFormData` for better readability.
Same for `setformErrors`